### PR TITLE
ISPN-13971 Add the Netty IOUring transport

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-13971

### CODE REVIEW

**Review by pruivo — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -380,7 +380,7 @@
       <versionx.org.jgroups.kubernetes.jgroups-kubernetes>2.0.1.Final</versionx.org.jgroups.kubernetes.jgroups-kubernetes>
       <versionx.org.jsoup>1.14.2</versionx.org.jsoup>
       <versionx.org.kohsuke.metainf-services.metainf-services>${version.metainf-services}</versionx.org.kohsuke.metainf-services.metainf-services>
-      <versionx.org.latencyutils.LatencyUtils>2.0.2</versionx.org.latencyutils.LatencyUtils>
+      <versionx.org.latencyutils.LatencyUtils>2.0.3</versionx.org.latencyutils.LatencyUtils>
```
Comment (wburns) [inline pom.xml:383]: Is this related?

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/NativeTransport.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+         if (IS_LINUX) {
+            HOTROD.ioUringNotAvailable(e.getMessage());
+         }
+      }
+      return false;
+   }
+
+   public static Class<? extends SocketChannel> socketChannelClass() {
+      if (USE_NATIVE_IOURING) {
+         return Util.loadClass("io.netty.incubator.channel.uring.IOUringSocketChannel", NativeTransport.class.getClassLoader());
```
Comment (wburns) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/NativeTransport.java:69]: I would suggest adding one more class that encapsulates all the IOUring references that is only loaded if USE_NATIVE_IORING is true. This way we have actual references to classes and get compilation errors if something is wrong or changes.

**Review by wburns — COMMENTED** [review]
LGTM just some minor things

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/NativeTransport.java (hunk 2 above)
Comment (tristantarrant) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/NativeTransport.java]: Done

File: pom.xml (hunk 1 above)
Comment (tristantarrant) [reply pom.xml:383]: Nope

**Comment by wburns** [comment]
> I have changed the order of preference: it's now Epoll, IOUring and NIO. Since IOUring is still in incubation, it's best if it is not the first choice.

I understand your reasoning. But I guess we should document how to enable IOURing as I would suspect we would want to have it enabled as much as possible.

**Comment by pruivo** [comment]
@tristantarrant `LicenseTest` is failing.  You need to add the ARM arch and io_uring to https://github.com/infinispan/infinispan/blob/fbf39a5acf379b3598e81e7b842cb6d0faf38e85/server/runtime/src/test/java/org/infinispan/server/license/LicenseTest.java#L91

**Comment by wburns** [comment]
Integrated into main, thanks @tristantarrant !
